DigitalLearningSolutions.Data.Migrations/202206281724_AddUniquenessConstraintsToCandidateAssessmentsAndSupervisorDelegates.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace DigitalLearningSolutions.Data.Migrations
+{
+    using FluentMigrator;
+    using FluentMigrator.SqlServer;
+
+    [Migration(202206281724)]
+    public class AddUniquenessConstraintsToCandidateAssessmentsAndSupervisorDelegates : Migration
+    {
+        private const string CandidateAssessmentsIndexName = "IX_CandidateAssessments_DelegateUserId_SelfAssessmentId";
+        private const string SupervisorDelegatesIndexName = "IX_SupervisorDelegates_DelegateUserId_SupervisorAdminId";
+
+        public override void Up()
+        {
+            Create.Index(CandidateAssessmentsIndexName).OnTable("CandidateAssessments").WithOptions()
+                .UniqueNullsNotDistinct().OnColumn("DelegateUserID").Ascending()
+                .OnColumn("SelfAssessmentID").Ascending();
+            Create.Index(SupervisorDelegatesIndexName).OnTable("SupervisorDelegates").WithOptions()
+                .UniqueNullsNotDistinct().OnColumn("DelegateUserID").Ascending()
+                .OnColumn("SupervisorAdminID").Ascending();

```

Comment:
I'm not sure what the usages of this table will be. Do we also want an index only on `SupervisorAdminID` only? For queries by `SupervisorAdminID` without querying by `DelegateUserID` I think this index won't be useful, will we encounter this case often?

Reply:
If the index is created as `(supervisorAdminId, delegateUserId)`, a query on `supervisorAdminId` only will be able to use the index. 

However, these indexes are both created with `DelegateUserID` first. In both cases, the order of the `OnColumn` statements should be reversed so that the indexes can be used in queries which only consider `selfAssessmentID` or `supervisorAdminId`.

Reply:
Good question, I should have made it clear: The index will not actually be referenced by anything (at least not that I'm aware of). It's a way to create the complex uniqueness constraint required by the ticket.

Reply:
We have existing queries with joins or where clauses on `supervisorAdminId` and `selfAssessmentID`. SQL server does not automatically add indexes on FK relationships, so we don't currently have them. Reversing the order of the columns in these new indexes therefore seems like a good way to kill two birds with one stone